ethereum/geth_client.go

```diff
@@ -19,15 +21,21 @@ import (
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/rpc"
 	logging "github.com/ipfs/go-log"
 )
 
-var log = logging.Logger("geth-client")
+var log = logging.Logger("ethereum")
 var gc Client
 var gcMu sync.RWMutex
 
+func init() {
+	gob.Register(common.Hash{})
```

Comment:
What is this for?

Reply:
gocelery uses go encoder to encode to persist. So apart from the primitive args, we should register the args that we send to gocelery v2